src/benchmarks/knucleotide.cpp

```diff
@@ -0,0 +1,128 @@
+// This code is adapted from Jeremy Zerfas [0] submission to the
+// "The Computer Language Benchmarks Game" [1], the appropriate license is at
+// the bottom of the file.
+//
+// [0] https://benchmarksgame-team.pages.debian.net/benchmarksgame/program/knucleotide-clang-1.html
+// [1] https://salsa.debian.org/benchmarksgame-team/benchmarksgame/
+
+
+#include <vector>
+#include <cstdint>
+
+#include "Map.h"
+#include "bench.h"
+
+
+// Macro to convert a nucleotide character to a code. Note that upper and lower
+// case ASCII letters only differ in the fifth bit from the right and we only
+// need the three least significant bits to differentiate the letters 'A', 'C',
+// 'G', and 'T'. Spaces in this array/string will never be used as long as
+// characters other than 'A', 'C', 'G', and 'T' aren't used.
+#define CODE_FOR_NUCLEOTIDE(nucleotide) (" \0 \1\3  \2"[nucleotide & 0x7])
+
+
+static size_t count(const std::vector<char> &poly, const std::string &oligo) {
+
+    using M = Map<uint64_t, uint32_t>;
+#ifdef USE_POOL_ALLOCATOR
+    Resource<uint64_t, uint64_t> resource;
```

Comment:
I think it should be t it be `Resource<uint64_t, uint32_t>`, or better `M::allocator_type::ResourceType`

Reply:
I've chosen to use `Resource<uint64_t, uint32_t>` for now, because it is more in line with the rest of the code.
I think it would be better to move it to `M::allocator_type::ResourceType` separately. It might even be a better idea to change the template arguments of `Resource` to the `Map` itself.